Machine.h

```diff
@@ -68,9 +69,10 @@ typedef struct Machine_ {
    UsersTable* usersTable;
    uid_t userId;
 
-   /* To become an array of lists - processes, cgroups, filesystems,... etc */
-   /* for now though, just point back to the one list we have at the moment */
-   struct ProcessList_ *pl;
+   unsigned int tableCount;
+   Table **tables;
```

Comment:
Because that'd be overkill? - same reason we don't use Vectors for the settings file and screens and other simple things.  We don't need complex operations on this array like insertion in the middle, sorting of elements, complex resizing with compaction, and so on.  It is just a simple old array that gets setup once and then occasionally walked left-to-right.